mlflow/java/client/src/main/java/org/mlflow/tracking/MlflowContext.java

```diff
@@ -0,0 +1,165 @@
+import org.mlflow.tracking.utils.DatabricksContext;
+import org.mlflow.tracking.utils.MlflowTagConstants;
+
+import java.util.*;
+import java.util.function.Consumer;
+
+/**
+ * Main entrypoint used to start MLflow runs to log to. This is a higher level interface than
+ * {@code MlflowClient} and provides convenience methods to keep track of active runs and to set
+ * default tags on runs which are created through {@code MlflowContext}
+ *
+ * On construction, MlflowContext will choose a default experiment ID to log to depending on your
+ * environment. To log to a different experiment, use {@link #setExperimentId(String)}} or
+ * {@link #setExperimentName(String)}
+ */
+public class MlflowContext {
+  private MlflowClient client;
+  private String experimentId;
+
+  /**
+   * Constructs a {@code MlflowContext} with a MlflowClient based on the MLFLOW_TRACKING_URI
+   * environment variable.
+   */
+  public MlflowContext() {
+    this(new MlflowClient());
+  }
+
+  /**
+   * Constructs a {@code MlflowContext} which points to the specified trackingUri.
+   *
+   * @param trackingUri The URI to log to.
+   */
+  public MlflowContext(String trackingUri) {
+    this(new MlflowClient(trackingUri));
+  }
+
+  /**
+   * Constructs a {@code MlflowContext} which points to the specified trackingUri.
+   *
+   * @param client The client used to log runs.
+   */
+  public MlflowContext(MlflowClient client) {
+    this.client = client;
+    this.experimentId = getDefaultExperimentId();
+  }
+
+  /**
+   * Returns the client used to log runs.
+   *
+   * @return the client used to log runs.
+   */
+  public MlflowClient getClient() {
+    return this.client;
+  }
+
+  /**
+   * Sets the experiment to log runs to by name.
+   * @param experimentName the name of the experiment to log runs to.
+   * @throws IllegalArgumentException if the experiment name does not match an existing experiment
+   */
+  public void setExperimentName(String experimentName) throws IllegalArgumentException {
+    Optional<Experiment> experimentOpt = client.getExperimentByName(experimentName);
+    if (!experimentOpt.isPresent()) {
+      throw new IllegalArgumentException(
+        String.format("%s is not a valid experiment", experimentName));
+    }
+    experimentId = experimentOpt.get().getExperimentId();
+  }
+
+  /**
+   * Sets the experiment to log runs to by ID.
+   * @param experimentId the id of the experiment to log runs to.
+   */
+  public void setExperimentId(String experimentId) {
+    this.experimentId = experimentId;
+  }
+
+  /**
+   * Returns the experiment ID we are logging to.
+   *
+   * @return the experiment ID we are logging to.
+   */
+  public String getExperimentId() {
+    return this.experimentId;
+  }
+
+  /**
+   * Starts a MLflow run. To log data to newly created MLflow run see the methods on
+   * {@link ActiveRun}. MLflow runs should be ended using {@link ActiveRun#endRun()}
+   *
+   * @param runName The name of this run.
```

Comment:
Maybe mention that this is used for display purposes only, and is stored in the `mlflow.runName` tag.